src/splice.ts

```diff
@@ -1,12 +1,133 @@
+import type {
+  ArraySplice,
+  GreaterThan,
+  IfNever,
+  IsLiteral,
+  IsNegative,
+  Subtract,
+  Sum,
+} from "type-fest";
+import type {
+  CoercedArray,
+  IterableContainer,
+  TupleParts,
+  WidenLiteral,
+} from "./internal/types";
 import { purry } from "./purry";
 
+type IfLiteral<N extends number, Fallback> = number extends N ? Fallback : N;
+
+type PositiveIndex<
+  T extends IterableContainer,
+  N extends number,
+> = N extends unknown
+  ? IsNegative<N> extends false
```

Comment:
This is just a nit but let's align with https://eslint.org/docs/latest/rules/no-negated-condition for types too and check the extends true branch first.